### DISCUSSION TITLE
building docs locally needs youtube api key

### DISCUSSION THREAD

**Discussion by c-nv-s** [saadeghi/daisyui#2607] [General]
should it be necessary to need a youtube api key in order to build the docs for local reference?
some people don't have continuous access to the internet thus building local docs is preferred... should sign up to google be a requirement to have access to daisyui docs locally?

**Comment by saadeghi** [comment]
It should be optional. Sorry.
If it gives error, add an empty
```
YOUTUBE_API_KEY=
```
to
```
src/docs/.env.local
```

I will update the docs today and make it optional.

**Reply by saadeghi** [reply]
Looks like it is optional already. I don't get any errors after removing the API key.

**Comment by c-nv-s** [comment]
still fails for me as there is no ``youtube.json`` file found
```
npm run get-json --silent && npm run createStyleHandlerComponent:prod --silent && NODE_ENV=production vite build --logLevel error

Error executing command: Error: Command failed: node --env-file=.env.local src/lib/scripts/get-youtube-resources.js 'src/lib/json/youtube.json'
node: bad option: --env-file=.env.local

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 9,
  killed: false,
  signal: null,
  cmd: "node --env-file=.env.local src/lib/scripts/get-youtube-resources.js 'src/lib/json/youtube.json'"
}
Error executing command: node: bad option: --env-file=.env.local

Error downloading JSON files: Error: Command failed: node --env-file=.env.local src/lib/scripts/get-youtube-resources.js 'src/lib/json/youtube.json'
node: bad option: --env-file=.env.local

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 9,
  killed: false,
  signal: null,
  cmd: "node --env-file=.env.local src/lib/scripts/get-youtube-resources.js 'src/lib/json/youtube.json'"

```

**Reply by c-nv-s** [reply]
@saadeghi sorry to pester but should I raise a separate ticket for the youtube.json file or can that file be added by you?

**Reply by saadeghi** [reply]
>```
>Error executing command: node: bad option: --env-file=.env.local
>```
I guess you need to [update Node.js](https://nodejs.org/en)

**Reply by c-nv-s** [reply]
looks like https://github.com/saadeghi/daisyui/blob/32744f9a1bd4afb191168fa631b375e99c3b9831/package.json#L47 needs to be changed to node 20 and above because it was failing with node 18 :disappointed:

**Reply by saadeghi** [reply]
daisyUI package is compatible with Node `16.9.0`  
daisyUI docs website [needs Node `20.6.0`](https://github.com/saadeghi/daisyui/blob/master/src/docs/package.json)
